client/src/components/DashboardCoponent/Content.tsx

```diff
@@ -0,0 +1,140 @@
+      setLength(12);
+    } else {
+      setLength(6);
+    }
+  };
+
+  return (
+    <>
+      <Grid container spacing={4}>
+        {length === 6 ? (
+          <>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+          </>
+        ) : (
+          <>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+            <Grid item xs={12} sm={6} md={4} xl={3}>
+              <Card />
+            </Grid>
+          </>
+        )}
+      </Grid>
+      {length === 6 ? (
+        <Box className={classes.btnContainer}>
```

Comment:
As opposed to rendering two different Buttons here, you can just add a ternary to the button text.
Eg: `{length === 6 ? "Show More" :  "Show Less"}`